src/btree/bt_cursor.c

```diff
@@ -2011,6 +2011,7 @@ __wt_btcur_range_truncate(WT_CURSOR_BTREE *start, WT_CURSOR_BTREE *stop)
     switch (btree->type) {
     case BTREE_COL_FIX:
         WT_ERR(__cursor_truncate_fix(start, stop, __cursor_col_modify));
+        *is_col_fix = true;
```

Comment:
It's good to have a boolean but I believe we can do `CUR2BT(start)->type == BTREE_COL_FIX` to check if we are dealing with FCS in `__wt_session_range_truncate`. It might be fewer changes?

Reply:
I think I tried doing something like that but was getting weird errors/segmentation faults, possibly because the B-tree didn't exist at the time, and also the B-tree could be NULL if there's no start or stop range specified.

Reply:
Another issue is that we may be truncating other types of data sources, not just files, so I'm not sure if we can directly case the cursor to a `WT_CURSOR_BTREE *`